main/acle.md

```diff
@@ -4726,6 +4881,2404 @@ assertions, which may speed up access to aligned data (and will fault
+
+> `svsetN` returns a copy of `tuple` in which the vector at index
+> `imm_index` has the value given by `x`. `imm_index` must be an
+> integer constant expression in the range [0, N-1].
+
+<!-- Trailing whitespace is deliberate here -->
+[`svget2(svBASEx2_t tuple, uint64_t imm_index)`](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svget2)  
+[`svget3(svBASEx3_t tuple, uint64_t imm_index)`](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svget3)  
+[`svget4(svBASEx4_t tuple, uint64_t imm_index)`](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svget4)
+
+> `svgetN` returns the value of vector `imm_index` in `tuple`. `imm_index`
+> must be an integer constant expression in the range [0, N-1].
+
+It is also possible to create completely-undefined vectors and tuples of
+vectors using [`svundef`](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svundef_),
+[`svundef2`](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svundef2),
+[`svundef3`](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svundef3) and
+[`svundef4`](https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=svundef4).
+
+### SVE predicate types
+
+[`<arm_sve.h>`](#arm_sve.h) defines a single sizeless predicate type
+`svbool_t`, which has enough bits to control an operation on a vector
+of bytes.
+
+The main use of predicates is to select elements in a vector. When the
+elements in the vector have *N* bytes, only the low bit in each sequence
+of *N* predicate bits is significant. For example:
+
+```{=html}
+<table>
+  <tr>
+    <th>Vector type</th>
+    <th colspan="9">Element selected by each <code>svbool_t</code> bit (from lsb)</th>
+  </tr>
+  <tr>
+    <td><code>svint8_t</code></td>
+    <td>0</td><td>1</td><td>2</td><td>3</td>
+    <td>4</td><td>5</td><td>6</td><td>7</td><td>8</td>
+  </tr>
+  <tr>
+    <td><code>svint16_t</code></td>
+    <td>0</td><td>-</td><td>1</td><td>-</td>
+    <td>2</td><td>-</td><td>3</td><td>-</td><td>4</td>
+  </tr>
+  <tr>
+    <td><code>svint32_t</code></td>
+    <td>0</td><td>-</td><td>-</td><td>-</td>
+    <td>1</td><td>-</td><td>-</td><td>-</td><td>2</td>
+  </tr>
+  <tr>
+    <td><code>svint64_t</code></td>
+    <td>0</td><td>-</td><td>-</td><td>-</td>
+    <td>-</td><td>-</td><td>-</td><td>-</td><td>1</td>
+  </tr>
+</table>
+```
+
+```{=tex}
+\begin{longtable}[]{@{}llllllllll@{}}
+\toprule
+Vector type & \multicolumn{9}{l}{Element selected by each \texttt{svbool\_t} bit (from lsb)}\tabularnewline
+\midrule
+\endhead
+\texttt{svint8\_t}  & 0 & 1 & 2 & 3 & 4 & 5 & 6 & 7 & 8\tabularnewline
+\texttt{svint16\_t} & 0 & - & 1 & - & 2 & - & 3 & - & 4\tabularnewline
+\texttt{svint32\_t} & 0 & - & - & - & 1 & - & - & - & 2\tabularnewline
+\texttt{svint64\_t} & 0 & - & - & - & - & - & - & - & 1\tabularnewline
+\bottomrule
+\end{longtable}
+```
```

Comment:
Can we try it as-is first, and consider something else if it doesn't work?  I tried both the pdf and html versions locally and it seemed to come out OK.

Reply:
> I tried both the pdf and html versions locally and it seemed to come out OK.
The HTML version is build with Jekyll, not pandoc . Or have you actually tested Jekyll locally?

Reply:
> > I tried both the pdf and html versions locally and it seemed to come out OK.
> 
> The HTML version is build with Jekyll, not pandoc . Or have you actually tested Jekyll locally?

I tried cut-&-pasting it into a github edit of README.md and clicking “Preview”.  It seemed to render well.

Reply:
> I tried cut-&-pasting it into a github edit of README.md and clicking “Preview”. It seemed to render well.

Oh, that's a very good data point, because I believe GitHub uses the same Jekyll both for browsing the project and for generating Pages.

You could always activate Pages on your github.io page and browse the final effect via https://rsandifo-arm.github.io/acle/

Reply:
I need complex(ish) tables for the SVE intrinsics documentation, so I looked for a way to keep the current HTML vs. LaTeX switch.  It turns out that kramdown doesn't support “```{=format}…```”, so the updated patch takes a different approach, described in the new covering message.  It seems to give good output for HTML and PDF, although TBH the HTML table doesn't look as good with the custom styling as it did with the default GitHub one.  We might want to tweak the style later (if only for this table).

Reply:
I've reworked the text and table so that we can use a simpler markdown table.